### PR TITLE
Add InferenceExecRequest.__repr__() for debug logging

### PR DESCRIPTION
**Pull request by renxida**
In the cycle leading up to the 3.2.0 release, we had some trouble triaging CI error reports / routing bugs to the appropriate teams / people. This and #917 are part of that.

The plan is to make shortfin's llm server log all relevant InferenceExecRequests when something (e.g. kvcache allocations / nans in logits) doesn't check out.

### CODE REVIEW

File: shortfin/tests/apps/llm/components/messages_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+@pytest.fixture
+def mock_request():
+    """
+    A mock InferenceExecRequest class that doesn't require worker context.
+
+    Useful for testing init and printing.
+    """
+    original_init = InferenceExecRequest.__init__
```
[the remark this answers is not in the document]
Comment (renxida) [reply shortfin/tests/apps/llm/components/messages_test.py]: Looked into the documentation link you sent.

Check out my new one-liner solution, where I don't even mock the __init__ - - just the VoidFuture which was causing InferenceExecRequest.__init__ to fail.

[the remark this answers is not in the document]
Comment (stbaione) [reply shortfin/tests/apps/llm/components/messages_test.py]: Nice, definitely more clear mocking the item that's specifically causing the error without it

**Review by stbaione — APPROVED** [review]
